packages/eslint-plugin/tests/rules/no-unnecessary-type-assertion.test.ts

```diff
@@ -25,10 +25,28 @@ if (
   const name = member.id as TSESTree.StringLiteral;
 }
     `,
+    `
+      const cx = 1;
+      let z0 = cx as number;
+      let z1 = cx as const;
+      let z2 = cx as 1;
+    `,
```

Comment:
```suggestion
    `
      const c = 1;
      let z = c as number;
    `,
    `
      const c = 1;
      let z = c as const;
    `,
    `
      const c = 1;
      let z = c as 1;
    `,
```

[Testing] _(nit pick)_ - https://github.com/typescript-eslint/typescript-eslint/pull/8558#discussion_r1510849264

Looks like this test case contains assertions for both literal and const assertions, what do you think about splitting them so that the assertions are more granular?